### PR TITLE
Appdev 8796 determine prefix from format and collection type

### PR DESCRIPTION
**Pull request by sharonluong**
In the call number generation, the code that finds the prefix needed depends on the prefix column in the formats table. This will need to change with the new prefix DB structure. This PR introduces a method that queries the DB for the right prefix to use instead.

### CODE REVIEW

File: app/Models/Prefix.php [hunk 1]
```diff
@@ -27,4 +28,34 @@ public function detachAllFormats()
...
   }
+
+  public static function findPrefixLabel($formatId, $collectionId)
+  {
+    $collectionTypeIdQuery = DB::table('collections')->select('collection_type_id')
+                                                     ->where('id', '=', $collectionId)
+                                                     ->get()
+                                                     ->first();
+
+    $collectionTypeId = $collectionTypeIdQuery === null ? null : $collectionTypeIdQuery->collection_type_id;
```
Comment (cazzerson) [inline app/Models/Prefix.php]: Throw exception (or allow PHP to) if the collection_type_id is missing/invalid

File: app/Models/Prefix.php [hunk 2]
```diff
@@ -27,4 +28,34 @@ public function detachAllFormats()
...
+    $labelQuery = DB::table('prefixes')->select('label')
+                                       ->join('format_prefix', 'prefixes.id', '=', 'format_prefix.prefix_id')
+                                       ->where([
+                                         ['format_prefix.format_id', '=', $formatId],
+                                         ['prefixes.collection_type_id', '=', $collectionTypeId]
+                                       ])
+                                       ->get()
+                                       ->first();
+
+    $label = $labelQuery === null ? null : $labelQuery->label;
```
Comment (cazzerson) [inline app/Models/Prefix.php]: Stop checking for null or throw exception for null once you're ready to remove the legacy code below.

File: tests/Unit/PrefixTest.php [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+use Illuminate\Foundation\Testing\DatabaseMigrations;
+use Illuminate\Foundation\Testing\DatabaseTransactions;
+
+class PrefixTest extends TestCase
+{
+  use DatabaseTransactions;
+
+  public function testFindPrefixLabel()
+  {
+    $this->disableExceptionHandling();
```
Comment (cazzerson) [inline tests/Unit/PrefixTest.php]: remove

File: tests/Unit/PrefixTest.php [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    $collection = factory(Collection::class)->create(['collection_type_id' => 3]);
+    $prefix1 = factory(Prefix::class)->create(['deleted_at' => null, 'collection_type_id' => 3]);
+    $prefix2 = factory(Prefix::class)->create(['deleted_at' => null, 'collection_type_id' => 4]);
+    $format1 = factory(Format::class)->create();
+    $format2 = factory(Format::class)->create();
+
+    // attach the formats to the prefix
+    $format1->prefixes()->attach($prefix1->id);
```
Comment (cazzerson) [inline tests/Unit/PrefixTest.php]: Add a couple of tests checking for expected exceptions for invalid data after earlier code is removed?

**Comment by sharonluong** [comment]
@cazzerson ready for re-review

**Pull request merged**